sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/dsl/package.scala

```diff
@@ -452,6 +453,16 @@ package object dsl {
 
       def hint(name: String, parameters: Any*): LogicalPlan =
         UnresolvedHint(name, parameters, logicalPlan)
+
+      def sample(
+          lowerBound: Double = 0.0,
+          upperBound: Double,
```

Comment:
It's pretty weird that a parameter without default value comes after a parameter with default value. If we want to follow Dataset, let's use overloads. Otherwise, let's make this `sample` method simple and good enough for the added tests.

Reply:
OK, let's keep `sample` simple.